### ISSUE TITLE
File listing regression in 2.3.6/2.3.7: Error when loading the folder

### ISSUE THREAD

**Issue by thegarbz** [haiwen/seadroid#1013]
I upgraded from 2.2.x recently and noticed a problem which I then hunted down by downgrading until the issue stopped. It seems between 2.3.5 and 2.3.6 something has changed which has caused problems with displaying libraries, and traversing folders. 

Expected behaviour: v2.3.5
Issue 1) Main libraries view sorted by most recent date. Libraries are shown with the most recent modification at the top.
Issue 2) Traversing folders: fast and snappy even for folders with large numbers of files.
Issue 3) (Related to Issue 2) Traversing large folder: fast and snappy. 

Actual behaviour: v.2.3.6/v2.3.7
Issue 1) Main libraries ignore sorting by more recent date. Sorting is set, and the dates are correctly shown, but the sorting seems completely random. 
Issue 2) Traversing folders: this seems to be highly dependent on the folder size. Small folders are okayish - still slower than 2.3.5. Large folders are painfully slow. It takes upwards of 20 seconds to display the folder contents of 5954 image files. On 2.3.5 this was <2seconds.
Issue 3) Traversing large folder: It appears to load for 30+ seconds and then briefly flashes the contents before display an error: "Error when loading the folder. Tap to retry". The working (but slow folder) has 5954 images in it. The next size up I have is 8236 images and this one fails with the error.

**Comment by SoLoR1** [comment]
Im getting "error when loading libraries", but i suspect its similar issue, since downgrading app to 2.3.5 fixes it, Its old self hosted seafile server (dont know version right now), that needs an update, but still...

**Comment by SoLoR1** [comment]
Also one more thing, from comments:
![image](https://github.com/haiwen/seadroid/assets/3218757/dedd2998-1f9f-4301-8073-fbaefe9ac766)

im not alone....

**Comment by toasterking** [comment]
I am also getting "Error when loading libraries" on seadroid 2.3.6. Can create a library but not see it. Downgrading to 2.3.5 fixes it. Seafile server version is 5.0.3. (Server is stuck at that version due to OS and hardware constraints. I mean, c'mon, it's only 9 years out of date!)

**Comment by Caseritto** [comment]
Came here to confirm the "error when loading libraries" error. Tried on 2.3.6 and 2.3.7. Rolling back to 2.3.5 let the libraries load.

**Comment by zhwanng** [comment]
Due to some modifications to the certificate verification method, the self-signed certificate may be abnormal. Please upgrade to the latest version and try again

**Comment by toasterking** [comment]
> Due to some modifications to the certificate verification method, the self-signed certificate may be abnormal. Please upgrade to the latest version and try again

Thanks for the response.  My self-signed certificate is accepted in seadroid 3.0.5 now, but it still fails to list the libraries.  Note the "Not found" flash message (circled) that appears when it attempts to list libraries.

![1](https://github.com/user-attachments/assets/448282e3-1f11-49dd-8af2-255591f7b73e)
![2](https://github.com/user-attachments/assets/15cc3926-c905-4a85-a083-5cc22a8482bc)
![3](https://github.com/user-attachments/assets/350a4b17-df74-418e-8857-ae5315baa211)
![4](https://github.com/user-attachments/assets/62d331c6-83e2-45c3-b437-6a50eec44437)
![5](https://github.com/user-attachments/assets/dc18acdd-67e0-4600-9c37-4ef1048bc204)
![6](https://github.com/user-attachments/assets/b70928e7-43db-45fa-9201-245e09c6af73)

**Comment by zhwanng** [comment]
Can you provide the log files of the app? log file path is: /sdcard/0/Android/com.seafile.seadroid/files/logs

**Comment by toasterking** [comment]
Certainly. There is not much in the log, but I will attach it here. This is from Seadroid 3.0.6, but the UI experience is still the same as in the screenshots previously posted.
[2025-01-16.log](https://github.com/user-attachments/files/18434333/2025-01-16.log)

**Comment by toasterking** [comment]
After installing the app and configuring it as above, that is the only log file in the logs directory.

**Comment by freeplant** [comment]
Please test again with the latest 3.0.11 version. If you still have the issue, please try to use a valid CA. If the issue persistent, please open another ticket and changing the version to 3.0.
